easybuild/easyconfigs/x/XALT/CC_config.py

```diff
@@ -0,0 +1,272 @@
+# If the value on the left is SKIP then if the path matches
+# the regular expression then executable is not acceptable so
+# no XALT tracking is done for that path.
+
+# This "table" is used to generate a flex routine that processes
+# the paths. So the regular express must follow flex rules.
+# In particular, in order to match the pattern must match the whole path
+# No partial matches allowed.  Also do not use $ to match the
+# end of the string.  Finally slash is a special character and must
+# be quoted with a backslash.
+
+# The path are conceptionally matched from the first regular
+# expression to the last.  Once a match is found no other later
+# matches are checked. The upshot of this is that if you want to
+# track say /usr/bin/ddt, but ignore everything in /usr, then keep
+# /usr/bin/ddt first and skip /usr/.* after.
+
+# If a path does not match any patterns it is marked as KEEP.
+
+# Programs like R, MATLAB and python* are marked as PKGS.  These programs
+# can optionally track the internal "import" that are used.
+
+path_patterns = [
+    # Track packages
+    ['PKGS',  r'.*\/R'],
+    ['PKGS',  r'.*\/MATLAB'],
+    ['PKGS',  r'.*\/python[0-9.]*'],
+
+    ['KEEP',  r'.*\/bin\/cp'],
+    ['KEEP',  r'.*\/bin\/mv'],
+    ['KEEP',  r'.*\/bin\/gawk'],
+    ['KEEP',  r'.*\/bin\/sed'],
+    ['KEEP',  r'.*\/bin\/perl'],
+    ['KEEP',  r'.*\/bin\/grep'],
+    ['KEEP',  r'.*\/bin\/bzip2'],
+    ['KEEP',  r'.*\/bin\/gzip'],
+    ['KEEP',  r'.*\/bin\/tar'],
+
+    ['SKIP',  r'^\/usr\/.*'],
+    ['SKIP',  r'^\/sbin\/.*'],
+    ['SKIP',  r'^\/bin\/.*'],
+    ['SKIP',  r'^\/etc\/.*'],
+    ['SKIP',  r'^\/root\/.*'],
+    ['SKIP',  r'\/cvmfs\/soft.computecanada.ca\/gentoo\/2020\/usr\/.*'],
+    ['SKIP',  r'\/cvmfs\/soft.computecanada.ca\/gentoo\/2020\/sbin\/.*'],
+    ['SKIP',  r'\/cvmfs\/soft.computecanada.ca\/gentoo\/2020\/bin\/.*'],
+    ['SKIP',  r'\/cvmfs\/soft.computecanada.ca\/gentoo\/2020\/etc\/.*'],
+    ['SKIP',  r'\/cvmfs\/soft.computecanada.ca\/gentoo\/2020\/root\/.*'],
+    ['SKIP',  r'\/cvmfs\/soft.computecanada.ca\/nix\/store\/.*'],
+
+    # General executables to ignore
+    ['SKIP',  r'.*\/git'],
+    ['SKIP',  r'.*\/lua'],
+    ['SKIP',  r'.*\/mpiCC'],
+    ['SKIP',  r'.*\/mpicc'],
+    ['SKIP',  r'.*\/mpicxx'],
+    ['SKIP',  r'.*\/mpif77'],
+    ['SKIP',  r'.*\/mpif90'],
+    ['SKIP',  r'.*\/mpifort'],
+    ['SKIP',  r'.*\/mpifc'],
+    ['SKIP',  r'.*\/mpigcc'],
+    ['SKIP',  r'.*\/mpigxx'],
+    ['SKIP',  r'.*\/mpiicc'],
+    ['SKIP',  r'.*\/mpiicpc'],
+    ['SKIP',  r'.*\/mpiifort'],
+    ['SKIP',  r'.*\/mpiexec.hydra'],
+    ['SKIP',  r'.*\/hydra_pmi_proxy'],
+    ['SKIP',  r'.*\/ompi_info'],
+    ['SKIP',  r'.*\/opal_wrapper'],
+    ['SKIP',  r'.*\/orterun'],
+    ['SKIP',  r'.*\/vtwrapper'],
+    ['SKIP',  r'.*\/conftest'],
+    ['SKIP',  r'.*\/CMakeTmp\/cmTryCompileExec[0-9][0-9]*'],
+    ['SKIP',  r'.*\/CMakeTmp\/cmTC_[a-f0-9][a-f0-9]*'],
+  ]
+
+# Remove tracking for these directories in these environments
```

Comment:
This worked for gentoo, but not nix because the files in the directories are symlinks. I decided to enumerate the specific directories excluded from gentoo, because it's more explicit and the nix counterpart can be put on one line (lines 66 to 71).